### ISSUE TITLE
Compatibility with recent Keras/Tensorflow (tf 2.0)

### ISSUE THREAD

**Issue by fqassemi** [cpury/keras_gradient_noise#2]
It does not work with the recent version of Keras/Tensorflow, apparently one has to add tensorflow before all keras.xxx, such as tensorflow.keras.optimizers!

**Comment by cpury** [comment]
Thanks for reporting this issue. While both `tf.keras` and `keras` exist at the moment, I expect `tf.keras` to be the future default choice. Let me see if I can make it work for both, otherwise, I will change it to use `tf.keras`.

**Comment by fqassemi** [comment]
Thanks. It failed for me after upgrading!

**Comment by cpury** [comment]
I haven't fixed it yet, but I think I got it working now. Will push in a bit and ping you then.

**Comment by cpury** [comment]
Ok it should work now. I hope I didn't break anything. Can you try `pip install -U keras_gradient_noise` and then rerunning your code?

**Comment by fqassemi** [comment]
Thanks. No worries, it did not break anything. I will double-check and let you know soon. BTW, since you are looking for cool projects you may look into this little program that I started to think about: https://github.com/fqassemi/PolyRegression. I intended to use your gradient_noise to avoid vanishing gradients that happen often working with linear activation and recurrent network.

**Comment by cpury** [comment]
@fqassemi were you able to test it? Good luck with your project, hope it works out :)

**Comment by statmlben** [comment]
It seems that "get_gradients" is not implemented in recent Keras / TF2. I just did a simple test: change the mean of the normal noise to a very big number, but it seems does not effect the updates. That is, in NoisyOptimizer class,

`            grads = [
                grad + K.random_normal(
                    grad.shape,
                    mean=100000000.0,
                    stddev=K.sqrt(variance),
                    dtype=K.dtype(grads[0])
                )
                for grad in grads
            ]
`
has same result with
`            grads = [
                grad + K.random_normal(
                    grad.shape,
                    mean=0.0,
                    stddev=K.sqrt(variance),
                    dtype=K.dtype(grads[0])
                )
                for grad in grads
            ]
`

**Comment by cpury** [comment]
Oh, that's not good! Thanks for investigating. I will have a look when I get the time.

**Comment by statmlben** [comment]
> Oh, that's not good! Thanks for investigating. I will have a look when I get the time.

Thanks. Pls let me know if anything I can help.
